### DISCUSSION TITLE
On "extract_buffered_coords"

### DISCUSSION THREAD

**Discussion by Crizzo98** [r-a-dobson/dynamicSDM#14] [Q&A]
Hi everyone, I got this error while running "extract_buffered_coords" function.

ERROR in Earth Engine servers: Can't get band number 0. Image has no bands.
ERROR :ee_monitoring was forced to stop before getting results
Error in `map()`:
ℹ In index: 1.
Caused by error in `.f()`:
! Can't `drive_get()` a file when `id` is `NA`.

Any clues on what I should do?
Thank you so much!

**Comment by r-a-dobson** [comment]
Hi there,

Could you send me some reproducible code for this error so that I can look into it? My email address is eerdo@leeds.ac.uk.

Thank you!
Rachel

**Comment by Crizzo98** [comment]
Dear Rachel,
thank you for your response.

I'll send you my code.
First, I created the moving window:

matrix <- get_moving_window(radial.distance = 4638.3,
                            spatial.res.degrees = 0.056,
                            spatial.ext = Italy_pol10$geometry)



"Italy_pol10" is a polygon of Italy from "rnaturalearth<https://www.rdocumentation.org/packages/rnaturalearth/versions/1.0.1>" package:


Italy_pol10<- ne_countries(country = "italy", scale=10)



Then, I ran the function



extract_buffered_coords(occ.data = complete_dataset_Pol,
                        datasetname = "GOOGLE/DYNAMICWORLD/V1",
                        bandname = c("water", "trees", "grass", "crops", "shrub_and_scrub", "built", "bare"),
                        spatial.res.metres = 10,
                        GEE.math.fun = "sum",
                        moving.window.matrix = matrix,
                        user.email = ***@***.***",
                        save.method = "combined",
                        temporal.level= "year",
                        agg.factor = 464,
                        varname = variablenames[7],
                        save.directory=extraction_directory_7)



I've already used my dataset to download other climatic variables with dSDM package so i don't think that data are the problem, but I attached them anyway.
I wasn't sure about the possibility to use "c" function in the argument "bandname", so I tried with a single band name but I got the same error.
I chose an agg. Factor of 464 because I'm matching climatic variables data with a resolution of 4638.3 meters.


Let me know if you have any other question.
Thank you for your time and have a good weekend.

Cristiano

________________________________
Da: radobson ***@***.***>
Inviato: venerdì 28 giugno 2024 11:41
A: r-a-dobson/dynamicSDM ***@***.***>
Cc: Cristiano VIZZO ***@***.***>; Author ***@***.***>
Oggetto: Re: [r-a-dobson/dynamicSDM] On "extract_buffered_coords" (Discussion #14)


Hi there,

Could you send me some reproducible code for this error so that I can look into it? My email address is ***@***.******@***.***>.

Thank you!
Rachel

—
Reply to this email directly, view it on GitHub<https://github.com/r-a-dobson/dynamicSDM/discussions/14#discussioncomment-9903642>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BF33K6R6T2CQV54TPP6SOFLZJU4XLAVCNFSM6AAAAABJ6CRE4SVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSMBTGY2DE>.
You are receiving this because you authored the thread.

Firma il tuo 5xmille all’Università di Parma. Aiutaci a potenziare la capacità di accoglienza, soprattutto abitativa, per le studentesse e gli studenti. - Indica 00308780345 nella tua dichiarazione dei redditi.
